### PR TITLE
Implement Camera2 - Image Capture Sample

### PR DESCRIPTION
**Pull request by madebymozart**
# Description
This CL contains sample code for Camera2 Image Capture. This sample demonstrate how a developer can capture and image using the Camera2 APIs. This sample also demonstrate howt o display the captured image via the `ImageViewerFragment` class

# How Has This Been Tested?
This sample app was tested on 3 devices on 3 different version of android:

**Test Configuration #1 - Pixel 6 Pro API 33 (Android 13)**
**Test Configuration #2 - Samsung Galaxy S22 Ultra (Android 12) **
**Test Configuration #3 - Pixel 5a With 5G (Android 11)**

# Checklist:
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

# Screenshots

### CODE REVIEW

File: samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt [hunk 1]
```diff
@@ -0,0 +1,640 @@
...
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+@Sample(
+    name = "Camera2 - Image Capture",
+    description = "This sample demonstrates how to capture and image and encode it into a JPEG "
+            + "container. Includes enable torch (flash) support is available.",
+    documentation = "https://developer.android.com/training/camera2/capture-sessions-requests",
+    owners = ["Mozart Louis (mozart@google.com)"],
```
Comment (marcelpinto) [inline samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt:63]: Let's avoid this feature. Better not to expose your email in the codebase + owners tend to change quite fast and we rather track this internally

File: samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt [hunk 2]
```diff
@@ -0,0 +1,640 @@
...
+import java.util.concurrent.ArrayBlockingQueue
+import java.util.concurrent.TimeoutException
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+import kotlin.coroutines.suspendCoroutine
+
+@Sample(
+    name = "Camera2 - Image Capture",
+    description = "This sample demonstrates how to capture and image and encode it into a JPEG "
+            + "container. Includes enable torch (flash) support is available.",
```
Comment (marcelpinto) [inline samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt:61]: "if it's available?"

**Review by marcelpinto — CHANGES_REQUESTED** [review]

File: samples/camera/camera2/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -18,6 +18,7 @@
 
     <uses-permission android:name="android.permission.CAMERA" />
     <uses-permission android:name="android.permission.RECORD_AUDIO" />
+    <uses-permission android:name="android.permission.POST_NOTIFICATIONS" />
```
Comment (marcelpinto) [inline samples/camera/camera2/src/main/AndroidManifest.xml:21]: I don't see it being used btw. Is it needed?

**Review by marcelpinto — CHANGES_REQUESTED** [review]

File: samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt (hunk 2 above)
Comment (madebymozart) [reply samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt:61]: Some Devices don't have a touch light (Flash Capabilities)

Comment (marcelpinto) [reply samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt:61]: Sure, but I mean the sentence is not correct. 

Includes enable torch (flash) support when available.

or 

Includes enable torch (flash) support if it's available.

Comment (madebymozart) [reply samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt:61]: Ah I see what you mean, Correcting in the next commit

File: samples/camera/camera2/src/main/AndroidManifest.xml (hunk 3 above)
Comment (madebymozart) [reply samples/camera/camera2/src/main/AndroidManifest.xml:21]: I originally meant for it to be used to send a quick notification so the user can view the photos in an external viewer, but not needed anymore

File: samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt [hunk 4]
```diff
@@ -0,0 +1,636 @@
...
+                        exif.saveAttributes()
+                        Log.d(TAG, "EXIF metadata saved: ${output.absolutePath}")
+                    }
+
+                    // Display the photo taken to user
+                    lifecycleScope.launch(Dispatchers.Main) {
+                        binding.fragmentCamera2ImageCaptureViewfinder.visibility = View.GONE
+                        binding.fragmentCamera2ImageCaptureButton.visibility = View.GONE
+
+                        parentFragmentManager.commit {
```
Comment (marcelpinto) [inline samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt:312]: The simple image viewer is already part of this fragment. You don't need the commit right?

File: samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt [hunk 5]
```diff
@@ -0,0 +1,636 @@
...
+                            binding.fragmentCamera2ImageCaptureImageView.setContent {
```
Comment (marcelpinto) [inline samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt:313]: How are you handling the back navigation? Once you set the content the image will take most of the screen and user cannot remove it right?

File: samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt [hunk 6]
```diff
@@ -0,0 +1,647 @@
...
+        binding.fragmentCamera2PostCaptureBackButton.visibility = View.VISIBLE
+        binding.fragmentCamera2ImageCaptureImageView.setContent {
+            SimpleImageViewer(location = imageLocation)
+        }
+    }
+
+    private fun returnToImageCapture() {
+        binding.fragmentCamera2ImageCaptureButton.visibility = View.VISIBLE
+        binding.fragmentCamera2PostCaptureBackButton.visibility = View.GONE
+        binding.fragmentCamera2ImageCaptureImageView.setContent {  }
```
Comment (marcelpinto) [inline samples/camera/camera2/src/main/java/com/example/platform/camera/imagecapture/Camera2ImageCapture.kt:233]: Nit: if you make it gone there is probably no need to set an empty content

**Review by marcelpinto — APPROVED** [review]

**Pull request merged**